### ISSUE TITLE
How to test of my test set after each epoch

### ISSUE THREAD

**Issue by da-orange-zi** [ultralytics/ultralytics#4283]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

If I may, I would like to inquire about a Python-related matter. I am currently attempting to incorporate a test of my test set after each epoch in my code. I've made an attempt by utilizing the validate function and modifying the parameters passed to it, but unfortunately, it does not seem to be functioning as intended.

I kindly request guidance on how to properly implement this functionality. How should I modify the code to achieve the desired outcome? Your assistance would be greatly appreciated.

To ensure that my inquiry is clear, I have provided a snippet of the code I've been working with:

### Additional

 def validate(self, flag):
        """
        Runs validation on test set using self.validator. The returned dict is expected to contain "fitness" key.
        """
        if flag == 'test':
            self.data['val'] = '/root/autodl-tmp/NEUDET_3.47_1_3part/test'
            self.args.data = '/root/autodl-tmp/ultralytics/yolo/cfg/data_NEUDET_turn_test.yaml'
            self.data['yaml_file'] = '/root/autodl-tmp/ultralytics/yolo/cfg/data_NEUDET_turn_test.yaml'
            
        metrics = self.validator(self)
        fitness = metrics.pop('fitness', -self.loss.detach().cpu().numpy())  # use loss as fitness measure if not found
        if not self.best_fitness or self.best_fitness < fitness:
            self.best_fitness = fitness

        self.data['val'] = '/root/autodl-tmp/NEUDET_3.47_1/train'
        self.args.data = '/root/autodl-tmp/ultralytics/yolo/cfg/data_NEUDET_turn.yaml'
        self.data['yaml_file'] = '/root/autodl-tmp/ultralytics/yolo/cfg/data_NEUDET_turn.yaml'

        return metrics, fitness

**Comment by glenn-jocher** [comment]
@da-orange-zi you probably want to use `split=test` instead of the default `split=val`. See https://docs.ultralytics.com/modes/val/#arguments
